File: apps/admin-console/src/types/categorization.ts
export interface PolicyBasisDTO {
  doc_id: string;
  title: string;
  score?: number;
}

export interface CategorizedTransactionDTO {
  txn_id: string;
  date: string;
  description: string;
  amount: number;
  category: string;
  confidence: number;
  section: string;
  auto_post: boolean;
  cited_rule?: string;
  rationale: string;
  policy_basis?: PolicyBasisDTO | null;
}

export interface CategorizationResponseDTO {
  total_count: number;
  auto_post_count: number;
  auto_post_rate: number;
  review_count: number;
  items: CategorizedTransactionDTO[];
}

File: backend/app/schemas/categorization.py
from typing import Optional, List
from pydantic import BaseModel

class PolicyBasisDTO(BaseModel):
    doc_id: str
    title: str
    score: Optional[float] = None

class CategorizedTransactionDTO(BaseModel):
    txn_id: str
    date: str
    description: str
    amount: float
    category: str
    confidence: float
    section: str
    auto_post: bool
    cited_rule: Optional[str] = "—"
    rationale: str
    policy_basis: Optional[PolicyBasisDTO] = None

class CategorizationResponseDTO(BaseModel):
    total_count: int
    auto_post_count: int
    auto_post_rate: float
    review_count: int
    items: List[CategorizedTransactionDTO]
